plugins/modules/jenkins_node.py

```diff
@@ -172,6 +240,376 @@
 IS_PYTHON_2 = sys.version_info[0] <= 2
 
 
+if sys.version_info[0] <= 3 or sys.version_info[1] < 8:
+    class cached_property(object):  # noqa
+        def __init__(self, func):
+            self.func = func
+
+        def __get__(self, instance, cls):
+            if instance is None:
+                return self
+
+            value = self.func(instance)
+            setattr(instance, self.func.__name__, value)
+
+            return value
+else:
+    from functools import cached_property
+
+
+def bool_to_text(value):  # type: (bool) -> str
+    return "true" if value else "false"
+
+
+def text_to_bool(text):  # type: (str) -> bool
+    if text == "true":
+        return True
+
+    if text == "false":
+        return False
+
+    raise ValueError("unexpected boolean text value '{}'".format(text))
+
+
+class Element:
+    def __init__(self, root):  # type: (et.Element | None) -> None
+        self.root = root
+
+    def get(self, key):  # type: (str) -> str | None
+        return None if self.root is None else self.root.get(key)
+
+    def set(self, key, value):  # type: (str, str) -> None
+        self.root.set(key, value)
+
+    def find(self, path):  # type: (str) -> et.Element | None
+        return None if self.root is None else self.root.find(path)
+
+    def remove(self, element):  # type: (et.Element) -> None
+        self.root.remove(element)
+
+    def append(self, element):  # type: (et.Element) -> None
+        self.root.append(element)
+
+    @property
+    def class_(self):  # type: () -> str | None
+        return self.get("class")
+
+    @class_.setter
+    def class_(self, value):  # (str) -> None
+        self.set("class", value)
+
+
+class Config:
+    @abstractmethod
+    def init(self):  # type: () -> et.Element
+        """Initialize XML element from config.
+
+        Returns:
+            Initialized XML element.
+        """
+        raise NotImplementedError
```

Comment:
Good point - I had wanted to use it to prevent instantiating instances of classes with unimplemented methods but I couldn't be bothered faffing about with a 2 + 3 compatible ABC/ABCMeta.

I will remove it.